src/main/java/apoc/meta/Meta.java

```diff
@@ -72,7 +107,13 @@ public static Types from(String typeName) {
             }
             return STRING;
         }
+
+        public static String inferType(List<?> list) {
+            Set<String> set = list.stream().map(e -> of(e).name()).collect(Collectors.toSet());
```

Comment:
This is potentially quite expensive (imagine a 10k sized list). We should probably limit it to `take(10)` or so.